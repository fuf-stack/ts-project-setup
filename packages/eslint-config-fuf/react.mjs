import { configs, rules } from 'eslint-config-airbnb-extended';
import jsxA11yPlugin from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooksPlugin from 'eslint-plugin-react-hooks';

import baseConfig from './base.mjs';
import fufStorybook from './fuf/fuf-react-storybook.mjs';
import fufReact from './fuf/fuf-react.mjs';

export default [
  // Base Config
  ...baseConfig,

  // Register React-related plugins explicitly for flat config resolution
  {
    name: 'fuf/react/plugins',
    plugins: {
      react: reactPlugin,
      'react-hooks': reactHooksPlugin,
      'jsx-a11y': jsxA11yPlugin,
    },
  },

  // Airbnb React Recommended Config
  ...configs.react.recommended,
  // Airbnb React TypeScript Config
  ...configs.react.typescript,
  // Strict React Config
  rules.react.strict,

  // FUF React Configs
  fufReact,
  fufStorybook,
];
