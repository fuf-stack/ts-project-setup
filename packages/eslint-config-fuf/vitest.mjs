// see: https://github.com/veritem/eslint-plugin-vitest

import vitest from '@vitest/eslint-plugin';

export default [
  {
    name: 'fuf/vitest',
    plugins: { vitest },

    // configure file patterns for tests
    files: ['**/*.spec.ts', '**/*.spec.tsx', '**/*.test.ts', '**/*.test.tsx'],

    // test file rules
    rules: {
      // recommended vitest rules
      ...vitest.configs.recommended.rules,

      // Turn off legacy import plugin rules if present upstream
      'import-x/extensions': 'off',
      'import/extensions': 'off',

      // Do not enforce production dependencies in tests
      'import/no-extraneous-dependencies': 'off',
      'import-x/no-extraneous-dependencies': 'off',
    },
  },
];
