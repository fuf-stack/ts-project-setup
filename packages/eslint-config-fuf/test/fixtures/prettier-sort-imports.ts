/* eslint-disable */

/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import-x/extensions */
/* eslint-disable import-x/no-unresolved */
/* eslint-disable no-unused-vars */

import moduleStyles from './Some.module.css';
import jsonFile from './data.json';
import fs from 'node:fs';
import react from 'react';

import relative from '../relative/path';
import png from './image.png';
import alsoAbsolute from 'src/also/absolute';
import someOtherThing from '@fuf-stack/some-other-thing';
import { describe, expect, it } from 'vitest';
import jpg from './image.jpg';
import path from 'node:path';

import svg from './image.svg';
import thirdParty from 'third-party-module';
import something from '@fuf-stack/something';
import absolute from 'src/absolute';
import type { ReactElement } from 'react';
import jpeg from './image.jpeg';

// INFO: css imports will not be touched
import 'some-module/style.css';

const MyComponent = (): ReactElement => null;

export default MyComponent;
