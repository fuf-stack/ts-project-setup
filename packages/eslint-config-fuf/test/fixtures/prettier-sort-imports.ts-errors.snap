[
  {
    "line": 7,
    "message": "Replace `moduleStyles·from·'./Some.module.css';` with `type·{·ReactElement·}·from·'react';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 8,
    "message": "Replace `jsonFile·from·'./data.json';` with `{·describe,·expect,·it·}·from·'vitest';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 10,
    "message": "Replace `react·from·'react` with `path·from·'node:path`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 12,
    "message": "Replace `lative·from·'../relative/path';` with `act·from·'react';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 13,
    "message": "Replace `png·from·'./image.png';⏎import·alsoAbsolute·from·'src/also/absolute';` with `thirdParty·from·'third-party-module';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 16,
    "message": "Replace `{·describe,·expect,·it·}·from·'vitest';` with `something·from·'@fuf-stack/something';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 17,
    "message": "Replace `jpg·from·'./image.jpg` with `absolute·from·'src/absolute`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 18,
    "message": "Replace `path·from·'node:path` with `alsoAbsolute·from·'src/also/absolute`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 20,
    "message": "Replace `svg·from·'./image.svg';` with `relative·from·'../relative/path';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 21,
    "message": "Replace `thirdParty·from·'third-party-module';` with `jsonFile·from·'./data.json';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 22,
    "message": "Replace `something·from·'@fuf-stack/somethin` with `jpeg·from·'./image.jpe`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 23,
    "message": "Replace `absolute·from·'src/absolute` with `jpg·from·'./image.jpg`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 24,
    "message": "Replace `type·{·ReactElement·}·from·'react';` with `png·from·'./image.png';⏎import·svg·from·'./image.svg';⏎`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 25,
    "message": "Replace `jpeg·from·'./image.jpeg` with `moduleStyles·from·'./Some.module.css`",
    "ruleId": "prettier/prettier",
  },
  {
    "line": 30,
    "message": "Expected block statement surrounding arrow body.",
    "ruleId": "arrow-body-style",
  },
]